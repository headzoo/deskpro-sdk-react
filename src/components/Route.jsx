import React from 'react';
import PropTypes from 'prop-types';

/**
 * Used by the Routes component to match components to route locations.
 *
 * Example:
 *
 * ```jsx
 * <Routes>
 *  <Route location="settings" component={PageSettings} />
 *  <Route location="index" component={PageIndex} />
 *  <Route defaultRoute component={PageNotFound} />
 * </Routes>
 * ```
 */
export default class Route extends React.Component {
  static propTypes = {
    location:     PropTypes.string,
    defaultRoute: PropTypes.bool,
    component:    PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
    children:     PropTypes.node
  };

  static defaultProps = {
    location:     '',
    children:     '',
    component:    '',
    defaultRoute: false
  };

  /**
   * @returns {string}
   */
  location = () => {
    return this.props.location;
  };

  /**
   * @returns {React.Component}
   */
  component = () => {
    return this.props.component;
  };

  /**
   * @returns {boolean}
   */
  defaultRoute = () => {
    return this.props.defaultRoute;
  };

  /**
   * @returns {XML}
   */
  render() {
    if (this.props.component) {
      return React.createElement(this.props.component);
    }
    return this.props.children;
  }
}
