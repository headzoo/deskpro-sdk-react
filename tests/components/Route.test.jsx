import React from 'react';
import renderer from 'react-test-renderer';
import Route from '../../src/components/Route';

it('Route renders its component', () => {
  const App = () => <div>Hello</div>;
  const renderedValue = renderer.create(
    <Route location="index" component={App} />
  ).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
