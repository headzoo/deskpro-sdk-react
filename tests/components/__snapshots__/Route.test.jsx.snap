// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route renders its component 1`] = `
<div>
  Hello
</div>
`;
